src/jit/importer.cpp

```diff
@@ -19377,3 +19377,69 @@ CORINFO_RESOLVED_TOKEN* Compiler::impAllocateToken(CORINFO_RESOLVED_TOKEN token)
     *memory                        = token;
     return memory;
 }
+
+//------------------------------------------------------------------------
+// SpillRetExprHelper: iterate through arguments tree and spill ret_expr to local varibales.
+//
+class SpillRetExprHelper
+{
+public:
+    SpillRetExprHelper(Compiler* comp) : comp(comp)
+    {
+    }
+
+    void StoreRetExprResultsInArgs(GenTreeCall* call)
+    {
+        GenTreePtr args = call->gtCallArgs;
```

Comment:
Do you need to do this for `gtCallObjp` too? If not, please add a comment explaining why.

Reply:
It is a good question. `this` can be a big tree in case of inlining and we can't guarantee that it doesn't include `ret_expr`. I will fix that.